### PR TITLE
Expose `end_timestamp` in `Span#finish` and `Transaction#finish`

### PR DESCRIPTION
**Pull request by sl0thentr0py**

### CODE REVIEW

**Review by AbhiPrasad — APPROVED** [review]

File: sentry-ruby/lib/sentry/span.rb [hunk 1]
```diff
@@ -90,11 +90,11 @@ def initialize(
 
     # Finishes the span by adding a timestamp.
     # @return [self]
-    def finish
+    def finish(end_timestamp: nil)
       # already finished
       return if @timestamp
```
Comment (st0012) [inline sentry-ruby/lib/sentry/span.rb:95]: This isn't mentioned in the spec so I want to double check it: we only allow the first override but not the second time or later?
If that's true, can we add a test case for it?

File: sentry-ruby/lib/sentry/span.rb [hunk 2]
```diff
@@ -90,11 +90,11 @@ def initialize(
 
     # Finishes the span by adding a timestamp.
     # @return [self]
```
Comment (st0012) [inline sentry-ruby/lib/sentry/span.rb:92]: We forgot to mention that it could return `nil` too (early return)

**Review by st0012 — COMMENTED** [review]
I think this can base on `master` instead.

File: sentry-ruby/lib/sentry/span.rb (hunk 1 above)
Comment (sl0thentr0py) [reply sentry-ruby/lib/sentry/span.rb:95]: ah no, it should override either way, good catch!

File: sentry-ruby/lib/sentry/span.rb (hunk 2 above)
Comment (sl0thentr0py) [reply sentry-ruby/lib/sentry/span.rb:92]: I think it should never return nil, will also fix

**Review by st0012 — APPROVED** [review]
I think we can merge this once the changelog entry is updated + the build is fixed by updating the base branch.

**Pull request merged**